src/persist-client/src/internal/metrics.rs

```diff
@@ -1976,6 +1979,38 @@ impl PushdownMetrics {
     }
 }
 
+#[derive(Debug)]
+pub struct ConsolidationMetrics {
+    pub(crate) parts_fetched: IntCounter,
+    pub(crate) parts_skipped: IntCounter,
+    pub(crate) parts_wasted: IntCounter,
+    pub(crate) join_latency_histogram: Histogram,
+}
+
+impl ConsolidationMetrics {
+    fn new(registry: &MetricsRegistry) -> Self {
+        ConsolidationMetrics {
+            parts_fetched: registry.register(metric!(
+                name: "mz_persist_consolidation_parts_fetched_count",
+                help: "count of parts that were fetched and used during consolidation",
+            )),
+            parts_skipped: registry.register(metric!(
+                name: "mz_persist_consolidation_parts_skipped_count",
+                help: "count of parts that were never needed during consolidation",
+            )),
+            parts_wasted: registry.register(metric!(
+                name: "mz_persist_consolidation_parts_wasted_count",
+                help: "count of parts that were fetched but not needed during consolidation",
+            )),
+            join_latency_histogram: registry.register(metric!(
```

Comment:
I'm passionate about this histogram, because I was referring to these stats even during testing.

For the flag in this PR we'll only end up running this code in this envd, but if we eg. roll out the streaming compaction flag we'd end up doing this all over the place.

Reply:
discussed this offline. I'm on board with a histogram for the total latency for executing an entire fast path peek. I'm less sold on needing one for join latency, but my objection is more "in principle" than "in practice", so if you end up feeling strongly on two histograms, that's fine